### PR TITLE
resolver: add comments to `Config` fields, and clean redundant yaml annotations

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: pkg/resolver/config.go [hunk 1]
```diff
@@ -9,23 +9,39 @@ import (
...
-	DisableAAAA bool     `yaml:"disable_aaaa,omitempty" toml:",omitempty" json:",omitempty"`
-	Iterative   bool     `yaml:"iterative,omitempty"    toml:",omitempty" json:",omitempty"`
-	Recursive   bool     `yaml:"recursive,omitempty"    toml:",omitempty" json:",omitempty"`
-	Servers     []string `yaml:"servers,omitempty"      toml:",omitempty" json:",omitempty"`
-	Suffixes    []string `yaml:"suffixes,omitempty"     toml:",omitempty" json:",omitempty"`
+	// DisableAAAA indicates this [Resolver] will discard AAAA entries
+	DisableAAAA bool `yaml:"disable_aaaa,omitempty" toml:",omitempty" json:",omitempty"`
 
-	Rewrites []RewriteConfig `yaml:"rewrite,omitempty" toml:",omitempty" json:",omitempty"`
+	// Iterative indicates this [Resolver] will go straight to the roots and
```
Comment (karasz) [inline pkg/resolver/config.go:28]: Iterative indicates this *that* [Resolver] will go straight to the DNS root servers and

File: pkg/resolver/config.go [hunk 2]
```diff
@@ -9,23 +9,39 @@ import (
...
+	// ask the authoritative servers for the answers.
+	Iterative bool `yaml:",omitempty" toml:",omitempty" json:",omitempty"`
+
+	// Recursive indicates this [Resolver] will ask servers to perform recursive lookups.
```
Comment (karasz) [inline pkg/resolver/config.go:32]: same as above

File: pkg/resolver/config.go [hunk 3]
```diff
@@ -9,23 +9,39 @@ import (
...
+	Recursive bool `yaml:",omitempty" toml:",omitempty" json:",omitempty"`
+
+	// Servers is a list of DNS servers to use for forwarding or iterative resolution.
+	// If this [Resolver] is designated as iterative and no servers are provided,
+	// a built-in list of root servers will be used.
```
Comment (karasz) [inline pkg/resolver/config.go:37]: a built-in list of root DNS servers will be used.

**Review by karasz — CHANGES_REQUESTED** [review]
LGTM maybe with some caveats? see comments

File: pkg/resolver/config.go (hunk 1 above)
Comment (karasz) [reply pkg/resolver/config.go:28]: > Iterative indicates this _that_ [Resolver] will go straight to the DNS root servers and

*that this*, sorry

**Comment by amery** [comment]
amended, please re-check @karasz

**Review by karasz — APPROVED** [review]
LGTM

**Pull request merged**